crates/scarb-api/src/artifacts/deserialized.rs

```diff
@@ -0,0 +1,39 @@
+use anyhow::{Context, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use serde::Deserialize;
+use std::fs;
+
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+pub(super) struct StarknetArtifacts {
+    pub version: u32,
+    pub contracts: Vec<StarknetContract>,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+pub(super) struct StarknetContract {
+    pub id: String,
+    pub package_name: String,
+    pub contract_name: String,
+    pub artifacts: StarknetContractArtifactPaths,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug, PartialEq, Clone)]
+pub(super) struct StarknetContractArtifactPaths {
+    pub sierra: Utf8PathBuf,
+}
+
+/// Get deserialized contents of `starknet_artifacts.json` file generated by Scarb
+///
+/// # Arguments
+///
+/// * `path` - A path to `starknet_artifacts.json` file.
+pub(super) fn artifacts_for_package(path: &Utf8Path) -> Result<StarknetArtifacts> {
+    let starknet_artifacts =
+        fs::read_to_string(path).with_context(|| format!("Failed to read {path:?} contents"))?;
+    let starknet_artifacts: StarknetArtifacts =
+        serde_json::from_str(starknet_artifacts.as_str())
+            .with_context(|| format!("Failed to parse {path:?} contents. Make sure you have enabled sierra code generation in Scarb.toml"))?;
```

Comment:
Rn it has nothing to do with sierra code gen anymore I think?

Reply:
It's tricky. Because this can be both sierra generation for older flow and just an error with test targets in new flow. I'll try to update this.